Health dynamic

Limalima64 opened the pull request:
- The UI of Health send a request with dynamic bucketName
- The server sends that bucketName to GetSDSHealth and then GetSDSHealth uses that to send a request to the minio

Limalima64 committed:
Update the Health dynamic with buckeName

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/AddBucketS3Config.tsx:
```diff
@@ -14,39 +14,174 @@
+            variant="contained"
+            tabIndex={-1}
+            onClick={() => {
+              const firstConfig: s3ConfigObject = {
```
do we really need that by default its an empty object ?

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/ComponentTabs.tsx:
```diff
@@ -0,0 +1,200 @@
+  handleErrorChange: (errors: boolean[]) => void;
+}
+
+const ComponentTabs: React.FC<ComponentTabsProps> = ({
```
naming could be improbed

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/ComponentTabs.tsx:
```diff
@@ -0,0 +1,200 @@
+  //Add a new tab
+  const handleAddTab = () => {
+    const newCurrentTabNumber = listS3Config.length;
+    const newConfigAdded: s3ConfigObject = {
```
empty object ?

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/ComponentTabs.tsx:
```diff
@@ -0,0 +1,200 @@
+
+      //Add or remove the secure key for the config
+      if (field === "secure") {
+        if (value) {
```
ternary operator

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/SelectedS3Config.tsx:
```diff
@@ -0,0 +1,100 @@
+  validInput: boolean;
+}
+
+const SelectedS3Config: React.FC<SelectedS3ConfigProps> = ({
```
naming

nic758 commented on portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket/addBucketThunks.ts:
```diff
@@ -69,22 +70,23 @@ export const addBucketAsync = createAsyncThunk(
+    if(listS3Config.length !== 0 && listS3Config != null){
+      const s3ConfigFile: { [key: string]: any } = {};
+
+      listS3Config.forEach((config, idx) => {
```
pretty sure you can use a much simpler approach

Limalima64 commented:
- removed the first config in the parent, now the S3ConfigTabs handle the first one
- renamed files
- changed ifs for ternary
- changed the approach in the addBucketThunks

MantleFounder commented:
/review

Limalima64 committed:
Small fix for health

Limalima64 commented:
- Verify params.Name is validated/sanitized and non-empty before passing it into GetSDSHealth
- Verify that useEffect depends on bucketName
- After a lot of testing the cancellation/race handling for rapid switches probleme doesnt happend. It shows the good health and not the bad one if we rapid switch
- For the Error/Empty State : We already have this for the errors : dispatch(setErrorSnackMessage(err));. And this possibility : "handle empty responses to avoid silent failures." its not possible because, we have a default configuration. Then none of the bucket has no storage for the health check-up
